### DISCUSSION TITLE
not sync some vm from vc

### DISCUSSION THREAD

**Discussion by ivannikita** [bb-Ricardo/netbox-sync#237] [General]
Hi! We have some vc cluster, run script to sync data, in cluster we have 600+ vm, in netbox we see 400+ vm, why script can not sync some vm?
setting.ini:
```ini
[netbox]
api_token = 
host_fqdn = netbox
port = 443
disable_tls = false
validate_tls_certs = false
prune_enabled = true
prune_delay_in_days = 30
default_netbox_result_limit = 200
timeout = 30
max_retry_attempts = 4
use_caching = true
[source/vsphere]
enabled = true
type = vmware
host_fqdn = vsphere
port = 443
validate_tls_certs = false
username = netbox
password = password
collect_hardware_asset_tag = True
dns_name_lookup = True
set_primary_ip = when-undefined
skip_vm_comments = False
sync_custom_attributes = True
```

**Comment by bb-Ricardo** [comment]
Hi,

When you sync, do you get any ERROR messages?

**Reply by ivannikita** [reply]
Hi,
I`m grep log for error and name of vm it is empty.

**Comment by bb-Ricardo** [comment]
Sorry, I can't really guess what is happening without any data.

**Comment by ivannikita** [comment]
What data you need?

**Comment by bb-Ricardo** [comment]
Can you provide the sync log with log level `DEBUG2` enabled?

**Reply by ivannikita** [reply]
Hi, logs below
[logs](https://drive.google.com/file/d/1yMk1ft1IDvE4LjFr05LTeEyzAgPvG2Bv/view?usp=sharing)

**Reply by ivannikita** [reply]
vm name  datahot01

**Reply by bb-Ricardo** [reply]
Hi, I can't find the vm name `datahot01` not in the logs. If this VM is managed by this vCenter then make sure the user `linux-netbox@virt.local` is able to read/see this object.
If this VM would be visible via API then you would at least see a `DEBUG: Parsing vCenter VM: datahot01` entry in the logs.

Also make sure to use the latest script version.

**Reply by ivannikita** [reply]
if i take info via API i see

```json   {
      "memory_size_MiB": 131072,
      "vm": "vm-393024",
      "name": "datahot01-FVgs",
      "power_state": "POWERED_ON",
      "cpu_count": 16
    },
```
in results

**Reply by ivannikita** [reply]
And in debug3 level log i found some inform about this vm but string `DEBUG: Parsing vCenter VM: datahot01` not include

**Reply by ivannikita** [reply]
Hi,
No ideas?

**Reply by bb-Ricardo** [reply]
Hi, It is difficult to debug this issue without any useful logs. And the API example does not look like a regular vCenter API.

you can also try to run the script with `-l DEBUG3` and write the output into a file. Then I could see If the vm is included in the vCenter output.

**Reply by ivannikita** [reply]
Hi,
[Link ](https://drive.google.com/file/d/19FT7y5VfAnc7Kicj_oz68na2cuw78Rs-/view?usp=sharing)to debug3 log

**Reply by bb-Ricardo** [reply]
I found the issue, the vm `uuid` exists multiple times. Only the first VM with this UUID is currently synced. A second VM with the same uuid is just ignored. There is another property `instanceUUID` which seems more unique. Will have to check how it is across different vCenter instances.

**Reply by ivannikita** [reply]
Hi,
How can we substitute  uuid with instanceUUID at settings?

**Reply by bb-Ricardo** [reply]
Hi, I or you have to update the code. Will try to push a updated version soon.

**Comment by bb-Ricardo** [comment]
Hi, I pushed a change to `development` branch. can you check it out and see if the VMs get synced now?

Thank you.

**Comment by ivannikita** [comment]
Hi,
We need some changes in settin.ini?

**Reply by bb-Ricardo** [reply]
nope, just make sure you are on branch `development`

**Comment by ivannikita** [comment]
run sync and see same picture of synced vm.

**Reply by bb-Ricardo** [reply]
can you run a `git log` command and post the output here?

**Comment by ivannikita** [comment]
Yea, i`am download not correct version. Now download correct, sync and all is ok. all vm is synced.

**Reply by bb-Ricardo** [reply]
great news. Then I will close the issue.

**Comment by ivannikita** [comment]
Hi, one more question.
When we sync vm, to his name added sufix like `data.somehost.com-UTZA`. how we can remove this prefix on sync?

**Reply by bb-Ricardo** [reply]
did you consult the settings-example.ini?

**Comment by ivannikita** [comment]
What parameter do you mean?

**Reply by ivannikita** [reply]
this strip_vm_domain_name = True ? his drop all inform aftter `data` in domain name `data.somehost.com-UTZA` and we have name like `data` but we need `data.somehost.com`

**Reply by bb-Ricardo** [reply]
sorry, this is not possible and won't be implemented.

**Reply by ivannikita** [reply]
Why? you split domain name by regexp? Maybe can change this regexp or put it in setting.ini?

**Reply by bb-Ricardo** [reply]
Well, this is an open source project. It is my free time I invest in this and don't get paid for it. I'm happy to contribute to the community as I use loads of amazing open sources tools every day as well.

If you really need this feature then you are more then welcome to add it yourself and create a Pull Request for merging. Or you have someone in your team who can add this functionality. Or you can hire a contractor to implement it. There are many ways.
